readthedocs/core/fields.py

```diff
@@ -0,0 +1,11 @@
+"""Shared model fields and defaults"""
+
+import sys
+from random import randrange
+
+from django.utils.http import int_to_base36
+
+
+def default_token():
+    """Generate default value for token field"""
+    return ''.join([int_to_base36(randrange(sys.maxint)) for _ in range(2)])
```

Comment:
This isn't a hash of any data in particular, so cryptographic security here isn't a concern. Randomization is the primary concern instead.

This was brought in from our auth token implementation, I'm not aware of a built-in token in Django. DRF uses this as a token generator:
https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/models.py#L37-L38

It might be best to replace this with the DRF version, as `randrange(sys.maxint)` has about `1:1000000000000000` chance of being 4 chars or less, or with concatenation of two of these strings, something like `1:1e-30` chance of being 8 characters or less.